orm/src/main/java/org/hibernate/search/jmx/impl/IndexControl.java

```diff
@@ -95,33 +96,27 @@ public void index(String entity) {
 			Thread.currentThread().interrupt();
 			throw new SearchException( "Unable to complete indexing" );
 		}
-		session.close();
 	}
 
 	@Override
 	public void optimize(String entity) {
 		Class<?> clazz = getEntityClass( entity );
-
 		SessionFactory factory = getSessionFactory();
-		Session session = factory.openSession();
-		FullTextSession fullTextSession = Search.getFullTextSession( session );
-		fullTextSession.beginTransaction();
-		fullTextSession.getSearchFactory().optimize( clazz );
-		fullTextSession.getTransaction().commit();
-		session.close();
+		ExtendedSearchIntegrator searchIntegrator = ContextHelper.getSearchIntegratorBySF( factory );
+		searchIntegrator.optimize( clazz );
 	}
 
 	@Override
 	public void purge(String entity) {
 		Class<?> clazz = getEntityClass( entity );
 
 		SessionFactory factory = getSessionFactory();
-		Session session = factory.openSession();
-		FullTextSession fullTextSession = Search.getFullTextSession( session );
-		fullTextSession.beginTransaction();
-		fullTextSession.purgeAll( clazz );
-		fullTextSession.getTransaction().commit();
-		session.close();
+		try ( Session session = factory.openSession() ) {
```

Comment:
Similar to the issue above, I think. Just in case the `FullTextSession` wrapper has resources of its own to close, we should do something like that:

```java
        try ( Session session = factory.openSession() ;
                FullTextSession fulltextSession = Search.getFullTextSession( session ) ) {
```